graphics/src/OBJLoader.cc

```diff
@@ -131,6 +131,50 @@ Mesh *OBJLoader::Load(const std::string &_filename)
             mat->SetTransparency(1.0 - m.dissolve);
             if (!m.diffuse_texname.empty())
               mat->SetTextureImage(m.diffuse_texname, path.c_str());
+
+            // load PBR textures
+            // Some obj exporters put PBR maps in the standard textures
+            // while others have proper support for obj PBR extension
+            Pbr pbrMat;
+            // PBR shoved into standard textures
+            if (!m.specular_texname.empty())
+              pbrMat.SetRoughnessMap(m.specular_texname);
+
+            // We use this field to determine if obj is exported by blender
+            // blender obj exporter puts roughness map in specular highlight
+            // field and metalness map in reflection map field!
+            // see summary in https://developer.blender.org/D8868
+            // detailing the existing exporter issues
+            // todo(anyone) remove this hack when blender fixes their exporter
+            // issue?
+            if (!m.specular_highlight_texname.empty())
+            {
+              pbrMat.SetRoughnessMap(m.specular_highlight_texname);
+              if (!m.reflection_texname.empty())
+                pbrMat.SetMetalnessMap(m.reflection_texname);
+            }
```

Comment:
I have mixed feelings about this, from one side it's nice to make blender assets work out of the box, but to be honest we might not need it for our use case since we are writing mtl and obj files manually and we have full control over the mtl fields.
Do you think for average users it is OK to have such hack to make Blender assets out of the box or do you think it's too "ugly"?

Reply:
yikes

I wonder if there's a way to detect if the file was created with Blender. It's nice to support Blender because it's widely used, but we shouldn't propagate their issues to our parser.

Reply:
ok I added a check for blender exported files. Looking at the obj files I have on disk, they have the exporter name in the first line of the .obj file so I'm using that to determine if the files are exported by blender or not. We'll likely need to add a blender version check to skip this workaround once the issue is fixed on their side. 96b109e